File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size()-1;
        while(l<=r){
            int mid = (int)(l+r)/2.0;
            if(target == nums[mid]) return mid;

            if(nums[l] <= nums[mid]){
                if(nums[l] > target || nums[mid] < target){
                    l = mid + 1;
                } else {
                    r = mid - 1;
                }
            } else {
                if(target < nums[mid] || target > nums[r]){
                    r = mid - 1;
                } else {
                    l = mid + 1;
                }
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n = nums.length;
        int l = 0;
        int sum = 0;
        int ans = Integer.MAX_VALUE;

        for (int r = 0; r < n; r++) {
            sum += nums[r];
            
            while (sum >= target) {
                ans = Math.min(ans, r - l + 1); // Record the minimum length
                sum -= nums[l];                 // Shrink the sum
                l++;                            // Move the left pointer
            }
        }
        
        return ans == Integer.MAX_VALUE ? 0 : ans;
    }
}